src/main/java/com/connexta/store/service/impl/StoreServiceImpl.java

```diff
@@ -195,23 +187,33 @@ public void addMetadata(String datasetId, List<MetadataInfo> metadataInfos) thro
               datasetId,
               Map.of());
           irmStorageAdaptor.updateStatus(datasetId, STORED);
-          indexDatasetClient.indexDataset(
-              datasetId,
-              UriComponentsBuilder.fromUri(storeUrl)
-                  .path(StoreController.RETRIEVE_DATA_URL_TEMPLATE)
-                  .build(datasetId, IRM_TYPE));
           break;
         default:
           throw new IllegalArgumentException(
               String.format("Received unsupported dataType {%s}", dataType));
       }
     }
+
+    unstage(datasetId);
+
+    indexClient.indexDataset(
+        UUID.fromString(datasetId),
```

Comment:
✏️ I prefer keeping the datasetId as a UUID always until we need it as a String so we don't need to validate it everywhere besides checking non-null. I suggest updating he method param to be a UUID.